datetime64 missing type overload for datetime.date

### Typing code example:

```python
import datetime
import numpy as np
d = np.datetime64(datetime.date(2021, 3, 17))
```

### Run mypy on file

```
untitled.py:3: error: No overload variant of "datetime64" matches argument type "date"  [call-overload]
untitled.py:3: note: Possible overload variant:
untitled.py:3: note:     def datetime64(self, Union[None, datetime64, Union[str, bytes], datetime] = ..., Union[Union[str, bytes], Tuple[Union[str, bytes], Union[int, integer[Any]]]] = ...) -> datetime64
scripts\untitled0.py:3: note:     <1 more non-matching overload not shown>
Found 1 error in 1 file (checked 1 source file)
```

Running the code with a datetime.datetime works fine, so it appears that numpy is missing datetime.date in the Union of allowable types to pass to np.datetime64

### NumPy/Python version information:

1.20.1 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)]

--- a/numpy/__init__.pyi
+++ b/numpy/__init__.pyi
@@ -2537,7 +2537,7 @@ class object_(generic):
 
 object0 = object_
 
-# The `datetime64` constructors requires an object with the three attributes below, 
+# The `datetime64` constructors requires an object with the three attributes below,
 # and thus supports datetime duck typing
 class _DatetimeScalar(Protocol):
     @property
@@ -2549,7 +2549,7 @@ class _DatetimeScalar(Protocol):
 
 
 class datetime64(generic):
-    @overload    
+    @overload
     def __init__(
         self,
         __value: Union[None, datetime64, _CharLike_co, _DatetimeScalar] = ...,

--- a/numpy/__init__.pyi
+++ b/numpy/__init__.pyi
@@ -2549,8 +2549,6 @@ class _DatetimeScalar(Protocol):
 
 
 class datetime64(generic):
-    #Replaced dt.datetime with _DatetimeScalar per issue #18640 and
-    #BvB93s recommendation
     @overload    
     def __init__(
         self,

--- a/numpy/__init__.pyi
+++ b/numpy/__init__.pyi
@@ -2537,8 +2537,8 @@ class object_(generic):
 
 object0 = object_
 
-#Added Protocol to typecheck for day, month, year property to be used in
-#datetime64 constructor per Issue #18640 and BvB93s recommendation
+# The `datetime64` constructors requires an object with the three attributes below, 
+# and thus supports datetime duck typing
 class _DatetimeScalar(Protocol):
     @property
     def day(self) -> int: ...

--- a/numpy/__init__.pyi
+++ b/numpy/__init__.pyi
@@ -2537,11 +2537,24 @@ class object_(generic):
 
 object0 = object_
 
+#Added Protocol to typecheck for day, month, year property to be used in
+#datetime64 constructor per Issue #18640 and BvB93s recommendation
+class _DatetimeScalar(Protocol):
+    @property
+    def day(self) -> int: ...
+    @property
+    def month(self) -> int: ...
+    @property
+    def year(self) -> int: ...
+
+
 class datetime64(generic):
-    @overload
+    #Replaced dt.datetime with _DatetimeScalar per issue #18640 and
+    #BvB93s recommendation
+    @overload    
     def __init__(
         self,
-        __value: Union[None, datetime64, _CharLike_co, dt.datetime] = ...,
+        __value: Union[None, datetime64, _CharLike_co, _DatetimeScalar] = ...,
         __format: Union[_CharLike_co, Tuple[_CharLike_co, _IntLike_co]] = ...,
     ) -> None: ...
     @overload

--- a/numpy/typing/tests/data/pass/scalars.py
+++ b/numpy/typing/tests/data/pass/scalars.py
@@ -84,6 +84,7 @@ np.datetime64(b"2019")
 np.datetime64("2019", "D")
 np.datetime64(np.datetime64())
 np.datetime64(dt.datetime(2000, 5, 3))
+np.datetime64(dt.date(2000, 5, 3))
 np.datetime64(None)
 np.datetime64(None, "D")
 

